### PR TITLE
[SDA-7847] Add OIDC attributes to tf provider ocm

### PR DESCRIPTION
**Pull request by gdbranco**
Related issue: https://issues.redhat.com/browse/SDA-7847
# What
Add attributes for BYO OIDC support through tf providder for ocm

# Why
Allow clients to use BYO OIDC

### CODE REVIEW

File: examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    export TF_VAR_url=...
+    ```
+
+* Decide STS operator_role_prefix    
+    ```
+    export TF_VAR_operator_role_prefix=...
+    ```
+
+and then run the `terraform apply` command.
+
```
Comment (bardielle) [inline examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/README.md:26]: Please add information on how to generate/ create your own OIDC, and what the values of `oidc_endpoint_url`, `oidc_private_key_secret_arn`, should be

File: examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/output.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+output cluster_id {
+  value = ocm_cluster_rosa_classic.rosa_sts_cluster.id
+}
+
+output oidc_thumbprint {
+  value = ocm_cluster_rosa_classic.rosa_sts_cluster.sts.thumbprint
+}
+
+output oidc_endpoint_url {
```
Comment (bardielle) [inline examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/output.tf:9]: not sure that it needs to be in the output file

File: examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/main.tf [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+locals {
+  sts_roles = {
+      role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/ManagedOpenShift-Installer-Role",
+      support_role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/ManagedOpenShift-Support-Role",
+      instance_iam_roles = {
+        master_role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/ManagedOpenShift-ControlPlane-Role",
+        worker_role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/ManagedOpenShift-Worker-Role"
+      },
+      operator_role_prefix = var.operator_role_prefix,
+      oidc_endpoint_url = "https://oidc-xxxx.s3.us-east-1.amazonaws.com",
```
Comment (bardielle) [inline examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/main.tf:44]: Please move it to the variable file : 
```suggestion
      oidc_endpoint_url = var.oidc_endpoint_url,
      oidc_private_key_secret_arn = 
```

File: provider/cluster_rosa_classic_resource.go [hunk 4]
```diff
@@ -506,6 +506,12 @@ func createClassicClusterObject(ctx context.Context,
 		instanceIamRoles.WorkerRoleARN(state.Sts.InstanceIAMRoles.WorkerRoleARN.Value)
 		sts.InstanceIAMRoles(instanceIamRoles)
 
+		if !state.Sts.OIDCEndpointURL.Unknown && !state.Sts.OIDCEndpointURL.Null && state.Sts.OIDCEndpointURL.Value != "" &&
```
Comment (bardielle) [inline provider/cluster_rosa_classic_resource.go:509]: I think you should split it - 
it the user set only one of those attributes maybe you should throw an error

File: examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/README.md (hunk 1 above)
Comment (gdbranco) [reply examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/README.md:26]: Those will come in a separate PR after I deal with STS module to include this possibility

File: examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/output.tf (hunk 2 above)
Comment (gdbranco) [reply examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/output.tf:9]: We can keep as a way to assure the client of what he's using. No need for it for other steps

File: examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/README.md (hunk 1 above)
Comment (gdbranco) [reply examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/README.md:26]: Got it now

File: examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/output.tf (hunk 2 above)
Comment (gdbranco) [reply examples/create_rosa_cluster/create_rosa_sts_cluster/byo_oidc_cluster/output.tf]: Removed :)

**Comment by bardielle** [comment]
I think you amend your commits into the merge commit...

**Comment by gdbranco** [comment]
> I think you amend your commits into the merge commit...

This is usually what we do, not sure why this got closed

**Pull request closed without merging**